logging/logging.go

```diff
@@ -0,0 +1,108 @@
+package logging
+
+import (
+	"log/slog"
+	"net/http"
+	"strings"
+	"time"
+
+	azcorePolicy "github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+var resourceTypes = map[string]bool{
+	"resourcegroups":   true,
+	"storageaccounts":  true,
+	"operationresults": true,
+	"asyncoperations":  true,
+}
+
+type LogRequestParams struct {
+	Logger    *slog.Logger
+	StartTime time.Time
+	Request   interface{}
+	Response  *http.Response
+	Error     error
+}
+
+// Shared logging function for REST API interactions
+func GetMethodInfo(method string, rawURL string) string {
+	url := strings.Split(rawURL, "?api-version")
+	parts := strings.Split(url[0], "/")
```

Comment:
Add check slice length check in case that the url is malformed.

Reply:
done